src/sysupdate/sysupdate-transfer.c

```diff
@@ -317,6 +317,8 @@ static int config_parse_resource_path(
                 return 0;
         }
 
+        // TODO: Here, try to substitute $BOOT with the boot mount point
```

Comment:
please don't do that here. instead add a new boolean field to `Resource` maybe called `rr->path_dollar_boot` or so, and then simply set that boolean here if the specified path starts with `^` (and unset the var if it doesnt start with `^`)